Realm/Realm/Realm.cs

```diff
@@ -1324,6 +1323,11 @@ public void RemoveAll()
         /// </summary>
         /// <remarks>
         /// The destination file cannot already exist.
+        ///
+        /// For synchronized realms, the file written will have the client file identifier removed.
+        /// Furthermore it is required that all local changes are synchronized with the server before the copy can be written.
+        /// This is to be sure that the file can be used as a starting point for a newly installed application.
+        /// The function will throw if there are pending uploads.
         /// <para/>
         /// If this is called from within a transaction it writes the current data, and not the data as it was when
```

Comment:
I know it's not in your PR, but I think the text here in the remark is a little confusing. Does it mean that it first will write the current data and only then do the copy? (So that the new changes are inlcuded)

Reply:
Also because of what you have added I suppose that this will throw an exception, right?

Reply:
It doesn't matter if I changed it or not. We want to make it as accurate and easy to understand as possible hence we need to update this. 👍 

The more I think about it and read the discussions in Slack about we should probably not state this part at all and rather make it clear that Realms should not be copied while anything is going on since that can lead to all kinds of problems (especially with Sync).

I'll re-write that part and then we can think about it and see if everything necessary is included.

Reply:
@papafe Update in bcbb402. What do you think?

Reply:
Seems fine, but I'm not sure about where the part with "If this is called from within a transaction it writes the current data, and no..." went. Is it nr.4 in the commit?

Reply:
https://github.com/realm/realm-dotnet/pull/2464#discussion_r682514879 addresses the same question. Let's continue there.